src/AddressRegistry.Api.Legacy/AddressMatch/Matching/StreetNameMatcher.cs

```diff
@@ -89,73 +85,112 @@ protected override AddressMatchBuilder DoMatchInternal(AddressMatchBuilder resul
             return results;
         }
 
-        protected override bool IsValidMatch(AddressMatchBuilder matchResult) => matchResult.AllStreetNames().Any();
+        protected override bool? IsValidMatch(AddressMatchBuilder? matchResult)
+            => matchResult?.AllStreetNames().Any();
+
+        protected override bool? ShouldProceed(AddressMatchBuilder? matchResult)
+            => IsValidMatch(matchResult);
 
-        protected override bool ShouldProceed(AddressMatchBuilder matchResult) => IsValidMatch(matchResult);
+        protected override IReadOnlyList<TResult>? BuildResultsInternal(AddressMatchBuilder? results)
+            => results?
+                .AllStreetNames()
+                .Select(w => _mapper.Map(w.StreetName))
+                .ToList();
 
         private void FindStreetNamesByKadStreetCode(AddressMatchBuilder results)
         {
-            foreach (var municipalityWrapper in results)
-            {
-                municipalityWrapper.AddStreetNames(_streetNameService.GetStreetNamesByKadStreet(results.Query.KadStreetNameCode, municipalityWrapper.NisCode));
-                //_telemetry.TrackStraatnaamMatch(gemeente.Naam, results.Query.Straatnaam, results.Query.KadStraatcode, results.Query.RrStraatcode, gemeente.Count(), (int)StraatnaamMatchLevel.KadStraatCode);
-            }
+            // TODO: Can't we skip this entire bit if there is no KadStreetNameCode query?
 
-            if (!string.IsNullOrEmpty(results.Query.StreetName) && !results.AllStreetNames().Any(w => w.StreetName.NameDutch.EqIgnoreCase(results.Query.StreetName)))
+            foreach (var municipalityWrapper in results)
```

Comment:
Can't we skip this entire bit if there is no `KadStreetNameCode` query?

Reply:
yep